src/Sarif.UnitTests/SarifVersionOneToCurrentTests.cs

```diff
@@ -317,60 +334,65 @@ public void SarifTransformerTests_ToCurrent_OneRunWitRules()
-            rule.Configuration.DefaultLevel.Should().Be(RuleConfigurationDefaultLevel.Error);
-            rule.HelpLocation.Should().NotBeNull();
-            rule.HelpLocation.Uri.ToString().Should().Be("http://www.domain.com/rules/c2002.html");
-            rule.MessageStrings.Should().BeNull();
-            rule.Properties.Should().BeNull();
-
-            rule = run.Resources.Rules["C2003"];
-
-            rule.Id.Should().Be("C2003");
-            rule.Name.Should().NotBeNull();
-            rule.Name.Text.Should().Be("Rule C2003");
-            rule.ShortDescription.Should().NotBeNull();
-            rule.ShortDescription.Text.Should().Be("Rules were meant to be broken.");
-            rule.FullDescription.Should().NotBeNull();
-            rule.FullDescription.Text.Should().Be("Rent internal rebellion competence biography photograph.");
-            rule.Configuration.Should().NotBeNull();
-            rule.Configuration.Enabled.Should().Be(false);
-            rule.Configuration.DefaultLevel.Should().Be(RuleConfigurationDefaultLevel.Note);
-            rule.HelpLocation.Should().BeNull();
-            rule.MessageStrings.Should().BeNull();
-            rule.Properties.Should().BeNull();
+            string v2LogText = JsonConvert.SerializeObject(v2Log, SarifTransformerUtilities.JsonSettingsV2);
+            string v2LogExpectedText =
+@"{
+  ""$schema"": ""http://json.schemastore.org/sarif-2.0.0"",
+  ""version"": ""2.0.0"",
+  ""runs"": [
+    {
+      ""tool"": {
+        ""name"": ""CodeScanner"",
+        ""semanticVersion"": ""2.1.0""
+      },
+      ""results"": [],
+      ""resources"": {
+        ""rules"": {
+          ""C2001"": {
+            ""id"": ""C2001"",
+            ""shortDescription"": {
+              ""text"": ""A variable was used without being initialized.""
+            },
+            ""messageStrings"": {
+              ""default"": ""Variable \""{0}\"" was used without being initialized.""
+            },
+            ""properties"": {
+              ""some_key"": ""FoxForceFive""
+            }
+          },
+          ""C2002"": {
+            ""id"": ""C2002"",
+            ""fullDescription"": {
+              ""text"": ""Catfish season continuous hen lamb include dose copy grant.""
+            },
+            ""configuration"": {
+              ""enabled"": true,
+              ""defaultLevel"": 2
+            },
+            ""helpLocation"": {
+              ""uri"": ""http://www.domain.com/rules/c2002.html""
+            }
+          },
+          ""C2003"": {
+            ""id"": ""C2003"",
+            ""name"": {
+              ""text"": ""Rule C2003""
+            },
+            ""shortDescription"": {
+              ""text"": ""Rules were meant to be broken.""
+            },
+            ""fullDescription"": {
+              ""text"": ""Rent internal rebellion competence biography photograph.""
+            },
+            ""configuration"": {}
```

Comment:
Why emit empty object?

Reply:
It's not actually empty. Enabled is false, and DefaultLevel is Note (0). So I guess those values lead to no output?

Reply:
Makes sense. I was doing the check but was looking for the wrong values.